### PR TITLE
Add bitcoin dev kit appendix

### PR DESCRIPTION
**Pull request by logosstone**
This new section covers:

- What is “Bitcoin Dev Kit”
- How to use the bdk-cli tool with examples
- Help documentation for bdk-cli tool commands and options

### CODE REVIEW

File: appdx-bitcoindevkit.asciidoc [hunk 1]
```diff
@@ -0,0 +1,709 @@
...
+| Pluggable blockchain clients +
+  * Compact block filters, single peer +
+  * Core RPC, cookie or passwd auth +
+|✓
+|
+|
+|
+
+| Pluggable coin selection +
+  * Custom user provided algorithm +
```
Comment (danielabrozzoni) [inline appdx-bitcoindevkit.asciidoc:129]: We also have the branch and bound algorithm and the single random draw :)

**Review by danielabrozzoni — COMMENTED** [review]
This looks really cool, good job!
The bdk release 1.0 shouldn't change too much of this file, I suppose just a couple of things on databases (sadly at the moment we're still not sure how they'll be structured, but probably a bit differently than now), so I'm happy that this appendix won't have to be completely re-written when that comes out!

**Comment by aantonop** [comment]
Thanks!

**Review by hernanmarino — COMMENTED** [review]
Great addition !

File: appdx-bitcoindevkit.asciidoc (hunk 1 above)
Comment (notmandatory) [reply appdx-bitcoindevkit.asciidoc:129]: > We also have the branch and bound algorithm and the single random draw :)

We have branch and bound on line 80 under Wallet functions. I don't think we currently have a way to do random draw on it's own, only as a fallback if there's no good branch and bound solution right? But maybe with the 1.0 changes that will be an option we can add.

Comment (danielabrozzoni) [reply appdx-bitcoindevkit.asciidoc:129]: Oh right, single random draw it's just a fallback. Yeah, this should change with bdk-core 1.0. But I really think it's important to add that we do have BnB, as the other two algorithms are really, really crappy privacy wise and fee wise...

Comment (danielabrozzoni) [reply appdx-bitcoindevkit.asciidoc:129]: Apparently I was misreading the document, bnb is mentioned above, sorry :)

**Pull request merged**